### DISCUSSION TITLE
GPIO on Raspberry Pi?

### DISCUSSION THREAD

**Discussion by wovo** [micropython/micropython#14161] [Unix & Windows]
Is there a way to access the GPIO from the Raspberry Pi(not the pico) MicroPython? 

For standard Python there is RPi.GPIO, but then I loose the MicroPython goodies like (soft) SPI & I2C, and framebuffer.

**Comment by jimmo** [comment]
@wovo Assume you mean using the Linux port of MicroPython, running on the RPi?

In which case I'm a bit confused by your question because the Linux port doesn't have SPI or I2C support either.

**Reply by cboostwvanooijen** [reply]
"the Linux port of MicroPython, running on the RPi"
correct

"because the Linux port doesn't have SPI or I2C support either"
I didn't know that, but I guess adding soft SPI and I2C would be simple. But useless without gpio, which is probably the resaon it is not supported now.

**Reply by KnorkFox** [reply]
Raspberry Pi, are famous for having gpio-ports. If it is easy to add SPI and I²C, it should be done. The original Linux of Raspberry (Raspian) even supports I²S!
This propably needs a native port of micropython on ARM 6 or ARM 7?

**Reply by peterhinch** [reply]
There are two issues here. As stated above, the Unix build of MicroPython will run on the Pi, but has no GPIO.

You are proposing a baremetal port. This has been proposed - the problem is that the Pi uses a Broadcom chip which is not ARM compatible (we already have ARM6 and ARM7 ports for RP2 and STM32).

For the moment the practical solution if to use CPython for Pi I/O.

**Reply by scruss** [reply]
> For the moment the practical solution if to use CPython for Pi I/O.

... and then fight over which of the many GPIO support libraries to use. Raspberry Pi have been terrible at supporting stable GPIO on their boards. They let single users' projects become standards, then when hardware changed, let those lone developers deal with the burden of the changes

**Comment by KnorkFox** [comment]
It does not need to be a product of Raspberry, but as a beginner, i can imagine that the border between "classic" microcontroller and small computer on a board, become more and more blurry with time, and 32 MB, 64 MB or 128 MB of ram, gets more popular in the future.
Media micro controller? Thanks, for your answers!
